### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by serentinos**
[DEMO LINK](https://serentinos.github.io/react_tabs-with-router/)

### CODE REVIEW

File: src/components/Tabs.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+interface Props {
+  tabs: Tab[],
+  selectedTabId?: string,
+}
+
+const getTabContent = (
+  tabsList: Tab[],
+  id: string,
+) => (
+  tabsList.find(tab => tab.id === id)?.content
```
Comment (Moroz-Dmytro) [inline src/components/Tabs.tsx:15]:
```suggestion
  tabsList.find(tab => tab.id === id)?.content ?? 'Please select a tab'
```

File: src/components/Tabs.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+              </li>
+            );
+          })}
+        </ul>
+      </div>
+
+      <div className="block" data-cy="TabContent">
+        {selectedTabId
+          ? getTabContent(tabs, selectedTabId)
+          : <p>Please select a tab</p>}
```
Comment (Moroz-Dmytro) [inline src/components/Tabs.tsx:49]: We can create a `const content = getTabContent(tabs, selectedTabId)` and just use it here

File: src/components/Tabs.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            return (
+              <li
+                key={id}
+                className={cn({ 'is-active': isSelected })}
+                data-cy="Tab"
+              >
+                <Link
+                  to={id}
+                  data-cy="TabLink"
+                  onClick={() => !isSelected}
```
Comment (Moroz-Dmytro) [inline src/components/Tabs.tsx:36]: What does this function do?

**Review by Moroz-Dmytro — CHANGES_REQUESTED** [review]
Good progress!
Added some coments

File: src/components/Tabs.tsx (hunk 3 above)
Comment (serentinos) [reply src/components/Tabs.tsx:36]: Nothing, this is rudimentary from prev task xD

**Comment by serentinos** [comment]
@Moroz-Dmytro Thank you for suggestions!

File: src/App.tsx [hunk 4]
```diff
@@ -1,51 +1,17 @@
...
-//   { id: 'tab-1', title: 'Tab 1', content: 'Some text 1' },
-//   { id: 'tab-2', title: 'Tab 2', content: 'Some text 2' },
-//   { id: 'tab-3', title: 'Tab 3', content: 'Some text 3' },
-// ];
+import { Outlet } from 'react-router-dom';
+import { NavBar } from './components/NavBar/NavBar';
 
 export const App = () => (
   <>
     {/* Also requires <html class="has-navbar-fixed-top"> */}
```
Comment (33Orange) [inline src/App.tsx:9]: Remove redundant comments

File: src/index.tsx [hunk 5]
```diff
@@ -1,10 +1,43 @@
...
+      },
+    ],
+  },
+]);
+
+createRoot(document.getElementById('root') as HTMLDivElement).render(
+  <React.StrictMode>
+    <RouterProvider router={router} />
+  </React.StrictMode>,
 );
```
Comment (33Orange) [inline src/index.tsx:43]: It's not the correct way to structure your application.
1. Let's use `App` as a root component, not as a layout
2. Create `AppRouter` that will render routing logic in your `app` component 
3. Create `Layout` component where you can render your view:
4. Remove the container to components to wrap them not in a react fragment

```
    <NavBar />

    <div className="section">
      <Outlet/>
    </div>
```

```suggestion
createRoot(document.getElementById('root') as HTMLDivElement).render(
  <React.StrictMode>
    <RouterProvider router={router} />
  </React.StrictMode>,
);
```

File: src/routes/HomePage.tsx [hunk 6]
```diff
@@ -0,0 +1,3 @@
+export const HomePage = () => (
+  <h1 className="title">Home page</h1>
+);
```
Comment (33Orange) [inline src/routes/HomePage.tsx:3]: Let's use our container here

```suggestion
export const HomePage = () => (
  <div className="container">
    <h1 className="title">Home page</h1>
  </div>
);
```

File: src/routes/NotFoundPage.tsx [hunk 7]
```diff
@@ -0,0 +1,3 @@
+export const NotFoundPage = () => (
+  <h1 className="title">Page not found</h1>
+);
```
Comment (33Orange) [inline src/routes/NotFoundPage.tsx:3]: Same

File: src/routes/TabsPage.tsx [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
+  return (
+    <>
+      <h1 className="title">Tabs page</h1>
+
+      <Tabs
+        selectedTabId={tabId}
+        tabs={tabs}
+      />
+    </>
```
Comment (33Orange) [inline src/routes/TabsPage.tsx:16]: Same about the container

**Review by 33Orange — CHANGES_REQUESTED** [review]
Almost done

A few moments to fix

**Pull request closed without merging**